src/test/scala/scalaz/stream/MergeSortedSpec.scala

```diff
@@ -0,0 +1,44 @@
+package scalaz.stream
+
+import org.scalacheck.Arbitrary._
+import org.scalacheck.{Gen, Arbitrary, Properties}
+import org.scalacheck.Prop._
+import scalaz.\&/
+import scalaz.std.anyVal._
+import scalaz.syntax.std.list._
+
+class MergeSortedSpec extends Properties("mergeSorted") {
+
+  case class Foo(i: Int, j: Int, s: String)
+  case class Bar(i: Int, j: Int, s: String)
+
+  implicit val arbitraryFoo = Arbitrary {
+    for {
+      (i, j, s) <- arbitrary[(Int, Int, String)]
+    } yield Foo(i, j, s)
+  }
+  implicit val arbitraryBar = Arbitrary {
+    for {
+      (i, j, s) <- arbitrary[(Int, Int, String)]
+    } yield Bar(i, j, s)
+  }
+
+  implicit val arbitraryFooInt: Arbitrary[Foo => Int] = Arbitrary { Gen.oneOf((_: Foo).i, (_: Foo).j) }
+  implicit val arbitraryBarInt: Arbitrary[Bar => Int] = Arbitrary { Gen.oneOf((_: Bar).i, (_: Bar).j) }
+
+  private def distinctOn[A, B](on: A => B): List[A] => List[A] = as => as.groupBy1(on).values.map(_.head).toList
+
+  property("") = forAll { (unsortedFoos: List[Foo], unsortedBars: List[Bar], f: Foo => Int, g: Bar => Int) =>
```

Comment:
At this point, we are only considering major bug fixes for inclusion in
0.8, and master is currently dead. But if you'd like to port this to the
new design, open up a PR against topic/redesign. Not sure on the naming and
location of this function but we could figure that out later.

Also since we aren't depending on scalaz, you can't use These or Order. :/
I'd just substitute the scala Ordering for Order, but not sure what to do
about These. I guess we could introduce that type to FS2 but that seems
silly just for this one function.

Somewhat related - it would be cool to have a module of functions for doing
various operations on sorted streams, including sorted merge, mergeBy,
inner join, outer join etc.
On Wed, Feb 17, 2016 at 5:42 PM alexwei <notifications@github.com> wrote:

> In src/test/scala/scalaz/stream/MergeSortedSpec.scala
> <https://github.com/functional-streams-for-scala/fs2/pull/543#discussion_r53243923>
> :
>
> > +    for {
> > +      (i, j, s) <- arbitrary[(Int, Int, String)]
> > +    } yield Foo(i, j, s)
> > +  }
> > +  implicit val arbitraryBar = Arbitrary {
> > +    for {
> > +      (i, j, s) <- arbitrary[(Int, Int, String)]
> > +    } yield Bar(i, j, s)
> > +  }
> > +
> > +  implicit val arbitraryFooInt: Arbitrary[Foo => Int] = Arbitrary { Gen.oneOf((_: Foo).i, (_: Foo).j) }
> > +  implicit val arbitraryBarInt: Arbitrary[Bar => Int] = Arbitrary { Gen.oneOf((_: Bar).i, (_: Bar).j) }
> > +
> > +  private def distinctOn[A, B](on: A => B): List[A] => List[A] = as => as.groupBy1(on).values.map(_.head).toList
> > +
> > +  property("") = forAll { (unsortedFoos: List[Foo], unsortedBars: List[Bar], f: Foo => Int, g: Bar => Int) =>
>
> Oops, I'll fix it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/functional-streams-for-scala/fs2/pull/543/files#r53243923>
> .
>